src/coreclr/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.CoreCLR.cs

```diff
@@ -11,7 +11,15 @@ public abstract partial class EqualityComparer<T> : IEqualityComparer, IEquality
     {
         // To minimize generic instantiation overhead of creating the comparer per type, we keep the generic portion of the code as small
         // as possible and define most of the creation logic in a non-generic class.
-        public static EqualityComparer<T> Default { [Intrinsic] get; } = (EqualityComparer<T>)ComparerHelpers.CreateDefaultEqualityComparer(typeof(T));
+        public static EqualityComparer<T> Default
+        {
+            // NoInlining helps JIT to track get_Default() accross all phases.
```

Comment:
Would it be better for the JIT itself to mark the method as non-inlineable instead?

Reply:
Done.